### PR TITLE
Feat: creates BOW

### PR DESCRIPTION
**Pull request by malayjoshi13**

### CODE REVIEW

File: TfIdf_BOW_creator.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import csv
+import numpy as np
+from regex_extraction import word_processor
+import re 
+import enchant
+dict = enchant.Dict("en_US")
```
Comment (abecerra) [inline TfIdf_BOW_creator.py]: Please, move this line after all the imports.

File: TfIdf_BOW_creator.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+from nltk.stem import PorterStemmer
+ps = PorterStemmer()
```
Comment (abecerra) [inline TfIdf_BOW_creator.py:9]: And, please leave two lines after the imports

File: TfIdf_BOW_creator.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                if word not in non_oligo_vocab:
+                    processed_word = word_processor(word)
+                    if not processed_word: #if not oligo looking sequence then append that in vocab coz 
+                        # we want that are linked with oligo in a sentence, but are themselves not an oligo
+                        non_oligo_vocab.append(word)
+
+    return non_oligo_sentences, oligo_sentences, non_oligo_vocab, oligo_vocab
+
+
+# Creating an index for each word in our vocab.
```
Comment (abecerra) [inline TfIdf_BOW_creator.py:84]: could this be a one-liner with enumerate?
https://book.pythontips.com/en/latest/enumerate.html

File: TfIdf_BOW_creator.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    index_dict = {} 
+    i = 0
+    for word in vocab:
+        index_dict[word] = i
+        i += 1
+    return index_dict
+
+
+# Create a count dictionary to keep the count of the number of documents containing the given word
+def count_dict(sentences, vocab):
```
Comment (abecerra) [inline TfIdf_BOW_creator.py:95]: Python also provides a Counter class for this kind of tasks:
https://docs.python.org/3/library/collections.html#collections.Counter

File: TfIdf_BOW_creator.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        processed_sentence+=" "+processed_word
+    return processed_sentence
+
+# Create vocab
+def create_vocab(filename):
+    # open the file in read mode
+    read_file = open(filename, 'r')
+    
+    # creating dictreader object
+    file = csv.DictReader(read_file)
```
Comment (abecerra) [inline TfIdf_BOW_creator.py:35]: All these functionality probably could be simplified using the pandas library ...

File: TfIdf_BOW_creator.py [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    for word in vocab:
+        word_count[word] = 0
+        for sent in sentences:
+            if word in sent:
+                word_count[word] += 1
+    return word_count
+
+
+# Term Frequency
+def termfreq(sentence, word):
```
Comment (abecerra) [inline TfIdf_BOW_creator.py:106]: A one-liner using Counter, too?

File: TfIdf_BOW_creator.py (hunk 3 above)
Comment (malayjoshi13) [reply TfIdf_BOW_creator.py:84]: Enumerate is something different. It's used to get the index and corresponding list element. We are not creating any list here. We want all these values to be returned when it's parent function is called.

File: TfIdf_BOW_creator.py (hunk 6 above)
Comment (malayjoshi13) [reply TfIdf_BOW_creator.py:106]: Same goes with this. I think you are confusing 
for counter, value in enumerate(my_list):
    print counter, value
    
with 

sentence, word

In our case "sentence" and "word" are two entities. We can 't place them in a list and use enumerate. In above case "counter" basically gives index of "value" which is an element of a list. Both cases are different.

File: TfIdf_BOW_creator.py (hunk 1 above)
Comment (malayjoshi13) [reply TfIdf_BOW_creator.py]: Made changes

File: TfIdf_BOW_creator.py (hunk 2 above)
Comment (malayjoshi13) [reply TfIdf_BOW_creator.py:9]: Made changes

File: TfIdf_BOW_creator.py (hunk 5 above)
Comment (malayjoshi13) [reply TfIdf_BOW_creator.py:35]: Yes, both are similar approaches

File: TfIdf_BOW_creator.py (hunk 4 above)
Comment (malayjoshi13) [reply TfIdf_BOW_creator.py:95]: "count_dict" is just the name of the function, it's not related to the "counter" class.

**Pull request merged**